sim_pkg/__main__.py

```diff
@@ -6,85 +8,37 @@
 import socketserver
 import csv
 import numpy as np
-from init_pose import init
 import argparse
+from sim_pkg.config import Configuration
 
-# sim_process_ = 0 
 robot_processes_ = []
-# vis_processes_ = 0
 parser = argparse.ArgumentParser(description='Run the simulator')
 
-parser.add_argument('-fn', '--filename', type=str, help="Name of the file without the .py extension")
+parser.add_argument('-fn', '--filename', type=str,
+                    help="Name of the file without the .py extension")
 
 args = parser.parse_args()
 
-def write_to_csv(free_port):
-    """
-    Write avaliable port number to csv file
-    """
-    f = open('port.csv', 'w')
-    num = [free_port]
-    with f:
-        # create the csv writer
-        writer = csv.writer(f)
-
-        # write a row to the csv file
-        writer.writerow(num)
-
-    # close the file
-    f.close()
-
-def init_pos(swarmsize):
-    """
-    Initialize the position
-    """
-    x = [0]*swarmsize
-    y = [0]*swarmsize
-    theta = [0]*swarmsize
-    a_ids = [0]*swarmsize
-
-    x, y, theta, a_ids = init(swarmsize,x,y,theta,a_ids)
-    id = [i for i in range(swarmsize)]
-
-    data = np.array([id, x, y, theta, a_ids]).T
-    np.savetxt("init.csv", data, delimiter=",", fmt="%.2f",
-           header="ID, x, y, theta, a_ids")
-
-
 def run():
     """
     Opens all the programs involved
     """
 
-    # Finds the first available socket on the system
-    with socketserver.TCPServer(("localhost", 0), None) as s:
-        free_port = s.server_address[1]
-
-    # Writes the socket number to port.csv file 
-    write_to_csv(free_port)
-    
     # Read required parameters from config.json
+    config = Configuration.from_path('config.json')  # Shouldn't be hardcoded.
     path = str(Path(__file__).parent)
-    with open('config.json', 'r') as myfile:
-        data=myfile.read()
-    config_var = json.loads(data)
-    num = config_var["NUMBER_OF_ROBOTS"]
-    use_vis = config_var["USE_VIS"]
-    # Initialize the positions in a csv file 
-    init_pos(num)
     
-    # print(path)
     # Open the simulator in a process
     sim_process_ = subprocess.Popen(['python3','-O', 'simulator.py'],close_fds=True,cwd=path)
     time.sleep(1)
 
     # Open all the robots
-    for i in range(int(num)):
+    for i in range(config.num_robots):
         r_process = subprocess.Popen(['python2','-O', 'bootloader.py', '-fn', args.filename],close_fds=True,cwd=path)
         robot_processes_.append(r_process)
     # subprocess.Popen(['python2','user2.py'],close_fds=True,cwd=path)
     
-    if use_vis == 1:
+    if config.use_visualizer:
         vis_processes_ = subprocess.Popen(['python3','-O','visualization.py'],close_fds=True,cwd=path)
```

Comment:
Please do not do this
https://linux.die.net/man/3/cwd

The problem with monkey patching your `cwd` is that you are expecting files to exist in a particular order.

Imagine I'm a user. Here's what I want to do:
```bash
pip install coachbot-simulator
coachbot-simulator -c config.json
```

And I want it to work. You gave me a set of scripts, I wanted a program.

Generally software will be installed in `/usr/local/lib/python3.11/dist-packages/<name-of-package>`. The problem with monkeypatching `cwd` like this is you're forcing the new scripts to exist in `/usr/local/lib/python3.11/dist-packages/<name-of-package>` but clearly `config.json` is (in my use case) in `~`. This breaks your program.

Generally bad practice.